pylint: error seen on ubuntu20.04 images

```
************* Module main
operator/main.py:592:0: R1710: Either all return statements in a function should return an expression, or none of them should. (inconsistent-return-statements)
make: *** [Makefile:77: pylint] Error 8
```

This happened after CI shifted to ubuntu-latest (20.04).

--- a/.github/workflows/on-pr-submit.yml
+++ b/.github/workflows/on-pr-submit.yml
@@ -32,6 +32,8 @@ jobs:
   run-on-minikube:
     name: kube 1.20.0
     runs-on: ubuntu-latest
+    env:
+      KUBE_VERSION: v1.20.0
     # Skip based on commit message
     if: "!contains(github.event.head_commit.message, '[ci skip]')"
     steps:
@@ -43,9 +45,21 @@ jobs:
     - name: Generate Manifests
       run: VERBOSE="true" make gen-manifest
     - name: Setup Minikube
-      run: tests/setup.sh v1.20.0
+      run: tests/setup.sh $KUBE_VERSION
     - name: Run tests
-      run: tests/travis-test.sh v1.20.0
+      run: tests/travis-test.sh $KUBE_VERSION
     - name: Cleanup
-      run: tests/cleanup.sh v1.20.0
+      run: tests/cleanup.sh $KUBE_VERSION
 
+      # Deploy KaDalu via Helm and run tests if PR address Helm charts
+    - name: Setup Minikube for Helm Charts
+      if: "contains(github.event.head_commit.message, '[helm]')"
+      run: tests/setup.sh $KUBE_VERSION
+    - name: Run tests
+      if: "contains(github.event.head_commit.message, '[helm]')"
+      run: tests/travis-test.sh $KUBE_VERSION
+      env:
+        INSTALL_TYPE: helm
+    - name: Cleanup
+      if: "contains(github.event.head_commit.message, '[helm]')"
+      run: tests/cleanup.sh $KUBE_VERSION

--- a/.github/workflows/on-release-tag.yml
+++ b/.github/workflows/on-release-tag.yml
@@ -62,6 +62,22 @@ jobs:
         overwrite: true
         file_glob: true
 
+  publish-helm-chart:
+    name: Publish Helm Chart
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Generate tgz archive of Helm Chart
+        run: |
+          KADALU_VERSION=$(echo "${{ github.ref }}" | sed -e 's,.*/\(.*\),\1,') make helm-chart
+      - name: Upload Helm Chart archive to release
+        uses: svenstaro/upload-release-action@v2
+        with:
+          repo_token: ${{ secrets.GITHUB_TOKEN }}
+          file: helm/kadalu/kadalu-helm-chart.tgz
+          overwrite: true
+          file_glob: true
+
   multi-arch-build-for-release:
     runs-on: ubuntu-latest
     steps:

--- a/Makefile
+++ b/Makefile
@@ -13,6 +13,7 @@ help:
 	@echo "    make release           - Publish the built container images"
 	@echo "    make prepare-release-manifests - Prepare release manifest files"
 	@echo "    make cli-build         - Build CLI binary"
+	@echo "    make helm-chart        - Create a tgz archive of Helm chart"
 
 build-grpc:
 	python3 -m grpc_tools.protoc -I./csi/protos --python_out=csi --grpc_python_out=csi ./csi/protos/csi.proto
@@ -114,6 +115,10 @@ pypi-build:
 	echo ${KADALU_VERSION} > server/VERSION
 	cd server; rm -rf dist; python3 setup.py sdist;
 
+helm-chart:
+	@echo "Creating tgz archive of helm chart(Version: ${KADALU_VERSION}).."
+	cd helm; tar -czf kadalu-helm-chart.tgz kadalu
+
 ifeq ($(TWINE_PASSWORD),)
 pypi-upload: pypi-build
 	cd server; twine upload --username kadalu dist/*
@@ -126,7 +131,7 @@ endif
 ifeq ($(KADALU_VERSION), latest)
 release: prepare-release
 else
-release: prepare-release pypi-upload cli-build
+release: prepare-release pypi-upload cli-build helm-chart
 	docker tag ${DOCKER_USER}/kadalu-operator:${KADALU_VERSION} ${DOCKER_USER}/kadalu-operator:${KADALU_LATEST}
 	docker tag ${DOCKER_USER}/kadalu-csi:${KADALU_VERSION} ${DOCKER_USER}/kadalu-csi:${KADALU_LATEST}
 	docker tag ${DOCKER_USER}/kadalu-server:${KADALU_VERSION} ${DOCKER_USER}/kadalu-server:${KADALU_LATEST}

--- a/cli/kubectl_kadalu/storage_add.py
+++ b/cli/kubectl_kadalu/storage_add.py
@@ -151,8 +151,10 @@ def get_kube_nodes(args):
         return nodes
     except utils.CommandError as err:
         utils.command_error(cmd, err.stderr)
+        return None
     except FileNotFoundError:
         utils.kubectl_cmd_help(args.kubectl_cmd)
+        return None
 
 
 def storage_add_data(args):

--- a/cli/kubectl_kadalu/storage_remove.py
+++ b/cli/kubectl_kadalu/storage_remove.py
@@ -72,6 +72,7 @@ def get_configmap_data(args):
 
     except utils.CommandError as err:
         utils.command_error(cmd, err.stderr)
+        return None
 
     except KeyError:
         # Validate method expects None when 'storage' not found.

--- a/helm/kadalu/Chart.yaml
+++ b/helm/kadalu/Chart.yaml
@@ -3,6 +3,6 @@ name: kadalu
 icon: https://kadalu.io/static/css/logo.png
 description: kadalu.io Storage Operator and CSI for k8s (with glusterfs)
 type: application
-version: 0.1.3
+version: 0.1.4
 
-appVersion: 0.7.3
+appVersion: 0.8.0

--- a/helm/kadalu/templates/clusterrole.yaml
+++ b/helm/kadalu/templates/clusterrole.yaml
@@ -24,7 +24,9 @@ rules:
       - pods
       - services
       - endpoints
+      - persistentvolumes
       - persistentvolumeclaims
+      - persistentvolumeclaims/status
       - customresourcedefinitions
       - events
       - configmaps
@@ -82,10 +84,13 @@ rules:
     verbs: ["get", "list", "watch", "create", "update", "patch"]
   - apiGroups: [""]
     resources: ["persistentvolumes"]
-    verbs: ["get", "list", "watch", "create", "delete"]
+    verbs: ["get", "list", "watch", "create", "delete", "patch"]
   - apiGroups: [""]
     resources: ["persistentvolumeclaims"]
-    verbs: ["get", "list", "watch", "update"]
+    verbs: ["get", "list", "watch", "update", "patch"]
+  - apiGroups: [""]
+    resources: ["persistentvolumeclaims/status"]
+    verbs: ["get", "list", "watch", "update", "patch"]
   - apiGroups: ["storage.k8s.io"]
     resources: ["storageclasses"]
     verbs: ["get", "list", "watch"]

--- a/operator/main.py
+++ b/operator/main.py
@@ -612,6 +612,7 @@ def get_configmap_data(volname):
             "Failed to get details from configmap",
             error=err
         ))
+        return None
 
 
 def get_num_pvs(storage_info_data):

--- a/templates/Replica2.client.vol.j2
+++ b/templates/Replica2.client.vol.j2
@@ -1,5 +1,6 @@
 volume {{ volume_id }}-ta
     type protocol/client
+    option transport.socket.read-fail-log false
     option remote-host {{ tiebreaker["node"] }}
     option remote-port {{ tiebreaker["port"] }}
     option remote-subvolume {{ tiebreaker["path"] }}
@@ -8,6 +9,7 @@ end-volume
 {% for brick in bricks %}
 volume {{ volname }}-client-{{ brick["brick_index"] }}
     type protocol/client
+    option transport.socket.read-fail-log false
     option volfile-key /{{ volname }}
     option remote-subvolume {{ brick["brick_path"] }}
     option remote-host {{ brick["node"] }}

--- a/templates/Replica3.client.vol.j2
+++ b/templates/Replica3.client.vol.j2
@@ -2,6 +2,7 @@
 volume {{ volname }}-client-{{ brick["brick_index"] }}
     type protocol/client
     option volfile-key /{{ volname }}
+    option transport.socket.read-fail-log false
     option remote-subvolume {{ brick["brick_path"] }}
     option remote-host {{ brick["node"] }}
 end-volume

--- a/templates/Replica3.shd.vol.j2
+++ b/templates/Replica3.shd.vol.j2
@@ -2,6 +2,7 @@
 volume {{ volname }}-client-{{ brick["brick_index"] }}
     type protocol/client
     option volfile-key /{{ volname }}
+    option transport.socket.read-fail-log false
     option remote-subvolume {{ brick["brick_path"] }}
     option remote-host {{ brick["node"] }}
 end-volume

--- a/tests/minikube.sh
+++ b/tests/minikube.sh
@@ -156,6 +156,8 @@ function install_kubectl() {
 # configure minikube
 MINIKUBE_VERSION=${MINIKUBE_VERSION:-"v1.15.1"}
 KUBE_VERSION=${KUBE_VERSION:-"v1.20.0"}
+HELM_VERSION=${HELM_VERSION:-"v3.5.0"}
+INSTALL_TYPE=${INSTALL_TYPE:-"default"}
 MEMORY=${MEMORY:-"3000"}
 VM_DRIVER=${VM_DRIVER:-"none"}
 #configure image repo
@@ -217,9 +219,16 @@ kadalu_operator)
 
     echo "Starting the kadalu Operator"
 
-    # pick the operator file from repo
-    sed -i -e 's/imagePullPolicy: Always/imagePullPolicy: IfNotPresent/g' manifests/kadalu-operator.yaml
-    kubectl apply -f manifests/kadalu-operator.yaml
+    if [[ "$INSTALL_TYPE" == "default" ]]
+    then
+      # pick the operator file from repo
+      sed -i -e 's/imagePullPolicy: Always/imagePullPolicy: IfNotPresent/g' manifests/kadalu-operator.yaml
+      kubectl apply -f manifests/kadalu-operator.yaml
+    elif [[ "$INSTALL_TYPE" == "helm" ]]
+    then
+      curl -fsSL -o get_helm.sh https://raw.githubusercontent.com/helm/helm/master/scripts/get-helm-3 && chmod +x get_helm.sh && ./get_helm.sh --version $HELM_VERSION
+      cd helm && helm install --create-namespace kadalu kadalu-chart kadalu/ --values kadalu/values.yaml && cd -
+    fi
 
     sleep 1
     # Start storage
@@ -299,7 +308,7 @@ clean)
 *)
     echo " $0 [command]
 Available Commands:
-  up               Starts a local kubernetes cluster and prepare disk for rook
+  up               Starts a local kubernetes cluster and prepare disks for gluster
   down             Stops a running local kubernetes cluster
   clean            Deletes a local kubernetes cluster
   ssh              Log into or run a command on a minikube machine with SSH

